File: bookstore-backend/src/main/java/com/bookstore/bookstore/presentation/response/BookResponse.java
package com.bookstore.bookstore.presentation.response;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.List;
import java.util.Map;
import java.util.UUID;

public record BookResponse(
        UUID id,
        String title,
        String isbn,
        String description,
        BigDecimal price,
        Integer stockQuantity,
        long soldCount,
        BigDecimal averageRating,
        long reviewCount,
        Map<Integer, Long> starBreakdown,
        String imageUrl,
        List<BookImageResponse> images,
        BookDetailResponse detail,
        UUID categoryId,
        UUID authorId,
        UUID publisherId,
        Instant createdAt,
        Instant updatedAt
) {
}

File: bookstore-desktop/Bookstore.Desktop/Dtos/BookResponse.cs
namespace Bookstore.Desktop.Dtos;

public class BookResponse
{
    public string Id { get; init; } = "";
    public string Title { get; init; } = "";
    public string? Isbn { get; init; }
    public decimal Price { get; init; }
    public int? StockQuantity { get; init; }
    public string? AuthorName { get; init; }
    public string? CategoryName { get; init; }
    public string? ImageUrl { get; init; }
}
